### ISSUE TITLE
POSIX & 'kill' - warn about non-portable usage

### ISSUE THREAD

**Issue by bittorf** [koalaman/shellcheck#1540]
#### For new checks and feature suggestions
- [x ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
kill -SIGUSR1 1234
kill -1 1234
```

#### Here's what shellcheck currently says:
no problems

#### Here's what I wanted or expected to see:
* do not prepend 'SIG' in symbols, so use USR1 instead of SIGUSR1
* use 'kill -s HUP' instead of 'kill -HUP'
* always use symbolic names (not kill -INTEGER PID)
 (the integers are platform-specific and can be seen with 'kill -l' or in signals.h)

**Comment by e-kwsm** [comment]
Some signals are specified in POSIX e.g. 9 (SIGKILL).

Following OP, diagnostic of the following script

```sh
kill -9 1
kill -7 1
kill -64 1

# `-1` is process group but not signal
kill -s TERM -1
kill -- -1

# definitely invalid
kill -s FOOBAR 1
```

would look like:

```

In - line 1:
kill -9 1
     ^-- SCXXXX(info): use `-s KILL` instead

In - line 2:
kill -7 1
     ^-- SCXXXX(error): signal 7 is unspecified in POSIX and not portable; use `-s BUS` or `-s EMT`

In - line 3:
kill -64 1
     ^-- SCXXXX(error): signal 64 is unspecified in POSIX and not portable

In - line 10:
kill -s FOOBAR 1
     ^-- SCXXXX(error): signal FOOBAR is unspecified in POSIX and not portable
```

- [kill(1P)](https://pubs.opengroup.org/onlinepubs/9799919799/utilities/kill.html)
- [&lt;signal.h&gt;](https://pubs.opengroup.org/onlinepubs/9799919799/basedefs/signal.h.html)

> The effects of specifying any <i>signal_number</i> other than those listed below are undefined.
